src/line/i-chartBuilder.test.js

```diff
@@ -0,0 +1,400 @@
+/**
+ * @jest-environment jsdom
+ */
+
+const fs = require("fs")
+const domTesting = require('@testing-library/dom')
+require('@testing-library/jest-dom')
+const userEvent = require("@testing-library/user-event").default
+
+//Spy on
+const window = require("./window")
+ 
+function initDomFromFiles(htmlPath, jsPath) {
+    window.localStorage.clear()
+    const html = fs.readFileSync(htmlPath, 'utf8')
+    document.open()
+    document.write(html)
+    document.close()
+    jest.isolateModules(function() {
+        require(jsPath)
+    })
+}
+
+/*
+    They should verify the behaviors described above. Note that these behaviors are 
+    duplicated across the three chart builder pages (bar.html, line.html, and scatter.html). 
+    You only need to test these behaviors on one of those pages.
+*/
+
+//ADDING VALUES TO THE CHART BUILDER
+
+test('Pressing "+" adds exactly one empty x and y value box', async function(){
+    initDomFromFiles(`${__dirname}/line.html`, `${__dirname}/line.js`)
+
+    //get + button for adding  new x and y cells.
+    const buttons = domTesting.queryAllByRole(document, "button")
+    const addButton = buttons[3];
+    
+    //make sure the correct button is selected
+    expect(addButton).toHaveTextContent("+")
+
+    //count all cells
+    var xValues = await domTesting.getAllByLabelText(document, "X")
+    var yValues = await domTesting.getAllByLabelText(document, "Y")
+
+
+    // Assert that the lists are not empty
+	expect(xValues).toHaveLength(1)
```

Comment:
There's some strange indentation on line 48; could you make sure it's indented with 4 spaces like the rest of the code?

Reply:
I still see different indentation on this line; could you please update it?

Reply:
Sorry, line 38 - the one right above this comment.